### PR TITLE
Added catpacks tests

### PR DESCRIPTION
**Pull request by Trial97**
Parent PR: #1253 

fixed some of my errors from the previous PR:https://github.com/PrismLauncher/PrismLauncher/pull/1253
I should have added tests from start(they also serve as example)

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -56,3 +56,6 @@ ecm_add_test(Index_test.cpp LINK_LIBRARIES Launcher_logic Qt${QT_VERSION_MAJOR}:
 
 ecm_add_test(Version_test.cpp LINK_LIBRARIES Launcher_logic Qt${QT_VERSION_MAJOR}::Test
     TEST_NAME Version)
+
+ecm_add_test(CatPack_test.cpp LINK_LIBRARIES Launcher_logic Qt${QT_VERSION_MAJOR}::Test
+    TEST_NAME CatPack)
```
Comment (Scrumplex) [inline tests/CMakeLists.txt]: missine newline :eyes:

**Review by getchoo — APPROVED** [review]
lgtm. the only (optional) change i'd suggest is if we're going to use the test file as an example, it might be nice to have it as a standalone file so it's a bit easier to read - though that could just go on our wiki page :p

**Comment by Trial97** [comment]
My intention was to have it there as an example until 8.0 was released and then add the documentation on the wiki page.

**Comment by TheKodeToad** [comment]
shouldn't it be using `testdata`?

**Review by ColonelGerdauf — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**